src/AzureAppConfigurationImpl.ts

```diff
@@ -34,20 +42,53 @@ export class AzureAppConfigurationImpl extends Map<string, unknown> implements A
 
         // Enable request tracing if not opt-out
         this.#requestTracingEnabled = requestTracingEnabled();
-        if (this.#requestTracingEnabled) {
-            this.#enableRequestTracing();
-        }
 
         if (options?.trimKeyPrefixes) {
             this.#sortedTrimKeyPrefixes = [...options.trimKeyPrefixes].sort((a, b) => b.localeCompare(a));
         }
+
+        if (options?.refreshOptions?.enabled) {

```

Comment:
Can we reuse `refreshEnabled()` method here?

Reply:
Technically we can, because `this.#refreshEnabled` depends on `this.#options` to have the correct result, and it's assigned at the beginning of the constructor. 
However, `this.#options` can be undefined according to its declaration, and TypeScript is not smart enough to know it's not undefined if we use `this.#refreshEnabled`. That would require we to add not-null assertion `!` as below:

```ts
// Currently type deduction works, because in the if block `options.refreshOptions` is not null.
if (options?.refreshOptions?.enabled) {
    const { watchedSettings, refreshIntervalInMs } = options.refreshOptions;

// After the change: no type deduction for this.#options, we have to add assertion, which is a little bit wired.
if (this.#refreshEnabled) {
    const { watchedSettings, refreshIntervalInMs } = this.#options!.refreshOptions!;   // <--  ! mark is not-null assertion
```

Reply:
Another reason is, `this.#refreshEnabled` is for future calls to know the enablement in elsewhere. Here in the constructor, we initialize related properties according to the passed parameter `options` instead of `this.#options`. It makes sense. 
So I prefer to keep it as is, although it looks like we can re-use it at a glance.